jujugui/static/gui/src/app/components/entity-content-readme/entity-content-readme.js

```diff
@@ -0,0 +1,106 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('entity-content-readme', function() {
+
+  juju.components.EntityContentReadme = React.createClass({
+
+    getInitialState: function() {
+      return {
+        readme: null,
+      };
+    },
+
+    componentDidMount: function() {
+      this._getReadme(this.props.entityModel);
+    },
+
+    /**
+      Get the content for the readme.
+
+      @method _getReadme
+      @param {Object} entityModel The entity model.
+      @return {Object} The readme content.
+    */
+    _getReadme: function(entityModel) {
+      var entityModel = this.props.entityModel;
+      var readmeFile = this._getReadmeFile(entityModel);
+      if (!readmeFile) {
+        this._getReadmeFailure();
+      } else {
+        var id = entityModel.get('id');
+        this.props.getFile(id, readmeFile, this._getReadmeSuccess,
+            this._getReadmeFailure);
+      }
+    },
+
+    /**
+      Get the filename for the readme.
+
+      @method _getReadmeFile
+      @param {Object} entityModel The entity model.
+      @return {Object} The readme filename.
+    */
+    _getReadmeFile: function(entityModel) {
+      var files = entityModel.get('files');
+      var match;
+      files.some(function(file) {
```

Comment:
This seems like an odd use of `some` to me. We're not really interested in the boolean returned by `some`, but rather it seems to be used in place of a vanilla for loop and a break statement.